src/Tools/Extensions.ApiDescription.Server/src/Microsoft.Extensions.ApiDescription.Server.csproj

```
@@ -19,7 +19,7 @@
       because publish output matches what was built.
     -->
     <Reference Include="dotnet-getdocument"
-      Condition=" '$(TargetFramework)' == 'netcoreapp2.1' "
+      Condition=" '$(TargetFramework)' != '$(DefaultNetFxTargetFramework)' "
```

This and https://github.com/dotnet/aspnetcore/pull/44129/files#diff-bef380c8e1f977d308eb069050a2c1da3ce3f17db75f3ab1f32d3624ecf85ba7R31 reverse the `Condition`. We intentionally published in the .NET Framework case because publication updates some files (compared to a regular build) and we need to ship those changes.

I suspect we need
```suggestion
      Condition=" '$(TargetFramework)' == '$(DefaultNetFxTargetFramework)' "
```

---

Hmmm. The comment on this reference seems to indicate the opposite.

```
<!--
      Publish the referenced projects when building .NET Core bits. This is not needed for full framework builds
      because publish output matches what was built.
    -->
```

From what I was observing, we needed to publish for both `netcoreapp2.1` and `net7.0`  because of the way the gathered the published output in the nuspec.

https://github.com/dotnet/aspnetcore/blob/ec4aa492ae450139a969b98eb63c6d9a91f70cdc/src/Tools/Extensions.ApiDescription.Server/src/Microsoft.Extensions.ApiDescription.Server.nuspec#L14-L15

If we used ` '$(TargetFramework)' == '$(DefaultNetFxTargetFramework)' ` we'd end up publishing for full framework builds but _not_ builds targeting .NET Core and the packaging would be borked.

---

Ignore me. Not enough espresso in the world I guess.